src/cmd.rs

```diff
@@ -0,0 +1,52 @@
+use std::path::Path;
+use std::{env, io};
+
+/// Read command and args.
+pub fn read_cmd() -> (usize, String) {
+    let mut input = String::new();
+    let c = io::stdin().read_line(&mut input).unwrap();
+    (c, input)
+}
+
+/// Execute a command with args.
+pub fn exec_cmd(cmd: &str, args: Vec<String>) -> io::Result<(i32, String, String)> {
+    if cmd == "cd" {
```

Comment:
why not using enums?

Reply:
using cost, as enum cannot have predef string and cannot use lib to extract variant name